src/aspire/abinitio/commonline_c3_c4.py

```diff
@@ -633,3 +783,110 @@ def _signs_times_v(self, vijs, vec):
             new_vec[ik] += s_ij_jk * vec[ij] + s_ik_jk * vec[jk]
 
         return new_vec
+
+    ###########################################
+    # Secondary Methods fo In-Plane Rotations #
+    ###########################################
+    @staticmethod
+    def complete_third_row_to_rot(r3):
+        """
+        Construct a rotation matrix whose third row is equal to the given row vector.
+
+        :param r3: A 1x3 vector of norm 1.
+        :return: A 3x3 rotation matrix whose third row is r3.
+        """
+
+        # If the third row coincides with the z-axis we return the identity matrix.
+        if norm(r3 - [0, 0, 1]) < 1e-5:
+            return np.eye(3)
+
+        # tmp is non-zero since r3 does not coincide with the z-axis.
+        tmp = np.sqrt(r3[0] ** 2 + r3[1] ** 2)
+
+        # Construct an orthogonal row vector of norm 1.
+        r1 = np.array([r3[1] / tmp, -r3[0] / tmp, 0])
+
+        # Construct r2 so that r3 = r1xr2
+        r2 = np.array([r3[0] * r3[2] / tmp, r3[1] * r3[2] / tmp, -tmp])
+
+        return np.vstack((r1, r2, r3))
+
+    @staticmethod
+    def clAngles2Ind(clAngles, n_theta):
+        thetas = np.arctan2(clAngles[:, 1], clAngles[:, 0])
+
+        # Shift from [-pi,pi] to [0,2*pi).
+        thetas = np.mod(thetas, 2 * np.pi)
+
+        # linear scale from [0,2*pi) to [0,n_theta).
+        return np.mod(np.round(thetas / (2 * np.pi) * n_theta), n_theta).astype(int)
+
+    @staticmethod
+    def g_sync(rots, order, rots_gt, dtype=np.float32):
```

Comment:
This is a symmetry based synchronization, so should probably stay here. 

just fyi: This synchronization finds the best `n` so that rotating the ground truth rotation by `2*pi*n/order` about the axis of symmetry best aligns with the estimated rotation. It then applies that rotation to the ground truth.

Reply:
Taking a look back at this comment... I think I misread `test_orient_symmetric` as `test_orient_sync`. Maybe this should be moved.